packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/ExtractionService.java

```diff
@@ -360,145 +362,81 @@ public void run() {
-				}
-			} catch (final FeignException e) {
-				final String error = "MIT get_mapping request failed";
-				throw new ResponseStatusException(
-						HttpStatus.valueOf(e.status()),
-						error + ": " + e.getMessage());
-			} catch (final Exception e) {
-				log.error("MIT merge failed", e);
-			}
-		}
+		// Set up the client interface
+		final String userId = currentUserService.get().getId();
+		final ClientEventInterface clientInterface = new ClientEventInterface(clientEventService, documentId, userId,
+			HALFTIME_SECONDS);
 
-		// add the attributes to the metadata
-		if (document.getMetadata() == null) {
-			document.setMetadata(new HashMap<>());
+		executor.execute(new Runnable() {
+			 @Override
+			 public void run() {
+
+				 clientInterface.sendMessage("Starting variable extraction.");
+				 try {
+					 // Fetch the text from the document
+					 DocumentAsset document = documentService.getAsset(documentId).orElseThrow();
+					 clientInterface.sendMessage("Document found, fetching text.");
+					 if (document.getText() == null || document.getText().isEmpty()) {
+						 throw new RuntimeException("No text found in paper document");
+					 }
+
+					 // add optional models
+					 final List<Model> models = new ArrayList<>();
+					 for (final UUID modelId : modelIds) {
+						 models.add(modelService.getAsset(modelId).orElseThrow());
+					 }
+					 clientInterface.sendMessage("Model(s) found, added to extraction request.");
+
+					 // Create a collection to hold the variable extractions
+					 JsonNode collection = null;
+
+					 clientInterface.sendMessage("Sending request to be processes by SKEMA and MIT.");
+					 final IntegratedTextExtractionsBody body = new IntegratedTextExtractionsBody(document.getText(), models);
+					 final ResponseEntity<JsonNode> resp = skemaUnifiedProxy.integratedTextExtractions(body);
+
+					 clientInterface.sendMessage("Response received.");
+					 if (resp.getStatusCode().is2xxSuccessful()) {
+						 for (final JsonNode output : resp.getBody().get("outputs")) {
+							 if (!output.has("errors") || output.get("errors").isEmpty()) {
+								 collection = output.get("data");
+								 break;
+							 }
+						 }
+					 } else {
+						 throw new RuntimeException("non successful response.");
+					 }
+
+					 if (collection == null) {
+						 throw new RuntimeException("No variables extractions returned");
+					 }
+
+					 clientInterface.sendMessage("Organizing and saving the extractions.");
+					 final List<JsonNode> attributes = new ArrayList<>();
+					 for (final JsonNode attribute : collection.get("attributes")) {
+						 attributes.add(attribute);
+					 }
+
+					 // add the attributes to the metadata
+					 if (document.getMetadata() == null) {
+						 document.setMetadata(new HashMap<>());
+					 }
+					 document.getMetadata().put("attributes", attributes);
+
+					 // update the document
+					 documentService.updateAsset(document).orElseThrow();
+
+				 } catch (final Exception e) {
```

Comment:
This will eat any exceptions thrown by the proxied services (skema unified, etc). Consider adding something like:

```
catch (final FeignException e) {
					final String error = "Transitive service failure";
					throw new ResponseStatusException(
							e.status()<100? HttpStatus.INTERNAL_SERVER_ERROR: HttpStatus.valueOf(e.status()),
							error + ": " + e.getMessage());
				}
```